src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java

```diff
@@ -103,6 +108,24 @@ private ItemStackMock(@NotNull ItemType type)
 
 	}
 
+	/**
+	 * Creates a new ItemStackMock instance if it isn't already mocked
+	 *
+	 * @param itemStack itemStack to be mocked
+	 * @return mocked itemStack
+	 */
+	public static ItemStackMock mock(ItemStack itemStack)
+	{
+		if (itemStack instanceof ItemStackMock itemStackMock)
+		{
+			return itemStackMock;
+		}
+		else
+		{
+			return new ItemStackMock(itemStack);
+		}
+	}
```

Comment:
This will either return a new instance of an item stack or not, this is generally going to cause issues, as you don't know whether the returned object is the same instance as the input object.

What you really need to do, is to use reflection to fetch the item stack mock field from item stack. Something like this when the item stack is not a mocked instance:
```java
Field field = ItemStack.class.getField("craftDelegate");
field.setAccessible(true);
return (ItemStackMock) field.get(itemStack);
```